### PR TITLE
feat(22-config): RemoteConfig integration

### LINKED ISSUE #22: As a client, I can configure the Storefront
On app startup, we need to load Firebase Remote Config with default or last fetched values, and then fetch the latest configuration.

### PR DESCRIPTION
**Pull request by ncuillery**

### CODE REVIEW

File: lib/product_detail/apis/product_api.dart [hunk 1]
```diff
@@ -1,18 +1,27 @@
...
 import 'package:get_it/get_it.dart';
 
-const PATH = '/occ/v2/electronics-spa/products/';
+const PATH = '/occ/v2/<CATALOG>/products';
 const PARAMS =
     '?fields=code,configurable,configuratorType,name,summary,price(formattedValue,DEFAULT),images(galleryIndex,FULL),baseProduct,averageRating,stock(DEFAULT),description,availableForPickup,url,numberOfReviews,manufacturer,categories(FULL),priceRange,multidimensional,tags&lang=en&curr=USD';
 
 class ProductApi {
   Future<Product> fetchProduct({required String productId}) async {
+    final path = PATH.replaceAll(
```
Comment (stephenrichter) [inline lib/product_detail/apis/product_api.dart:15]: Hmm maybe we move these paths to a utility class. I have an idea, we can discuss next meeting.

File: lib/init.dart [hunk 2]
```diff
@@ -28,4 +29,12 @@ void init() {
   GetIt.instance.registerSingleton(CategoryApi());
   GetIt.instance.registerSingleton(ProductApi());
   GetIt.instance.registerSingleton(ProductListApi());
+
+  GetIt.instance.registerSingletonAsync(() async {
```
Comment (stephenrichter) [inline lib/init.dart:33]: As mentioned previously, do we want to register this prior to our Api's (which rely on the path configs - in theory)?

**Review by stephenrichter — APPROVED** [review]
This looks good, a few comments for the future.

File: lib/init.dart (hunk 2 above)
Comment (ncuillery) [reply lib/init.dart:33]: Hmm right, I put it after without thinking a lot about it, about API < Repository < Bloc

But if the API relies on it, there is a dilemma 🤔

**Pull request merged**